src/modm/driver/dac/ma12070p.hpp

```diff
@@ -0,0 +1,354 @@
+
+#if MODM_HAS_IOSTREAM
+	friend IOStream&
+	operator << (IOStream& os, const ErrorRegister_t& c)
+	{
+		os << "ErrorRegister(";
+		if(c & ErrorRegister::FlyingCapOverVolt)
+			os << "FlyingCapOverVolt ";
+		if(c & ErrorRegister::OverCurrent)
+			os << "OverCurrent ";
+		if(c & ErrorRegister::Pll)
+			os << "Pll ";
+		if(c & ErrorRegister::PvddUnderVolt)
+			os << "PvddUnderVolt ";
+		if(c & ErrorRegister::OverTempWarning)
+			os << "OverTempWarning ";
+		if(c & ErrorRegister::OverTempError)
+			os << "OverTempError ";
+		if(c & ErrorRegister::PinToPinLowImpedance)
+			os << "PinToPinLowImpedance ";
+		if(c & ErrorRegister::DcProtection)
+			os << "DcProtection ";
+		os << ")";
+		return os;
+	}
+#endif
+
+
+	// quarter_decibel type
+	enum class
+	quarter_decibel_t : int16_t
+	{};
+
+	template<typename T>
+	static constexpr quarter_decibel_t
+	quarter_decibel(T value)
+	{
+		return quarter_decibel_t(int16_t(value * 4ul));
+	}
+
+	static constexpr float
+	quarterDecibelToFloat(quarter_decibel_t value)
+	{
+		return int16_t(value) / 4.f;
+	}
+
+	static constexpr quarter_decibel_t MaxVolume = static_cast<quarter_decibel_t>(24*4);
+	static constexpr quarter_decibel_t MinVolume = static_cast<quarter_decibel_t>(-144*4);
+	// static constexpr quarter_decibel_t MaxVolume = quarter_decibel(24);
+	// static constexpr quarter_decibel_t MinVolume = quarter_decibel(-144);
+};  // struct ma12070p
+
+constexpr ma12070p::quarter_decibel_t
+operator-(ma12070p::quarter_decibel_t v)
+{
+	return static_cast<ma12070p::quarter_decibel_t>(-static_cast<int16_t>(v));
+}
+
+namespace literals
+{
+	constexpr ma12070p::quarter_decibel_t
+	operator""_q_db(long double value)
+	{
+		return ma12070p::quarter_decibel(value);
+	}
+
+	constexpr ma12070p::quarter_decibel_t
+	operator""_q_db(unsigned long long int value)
+	{
+		return ma12070p::quarter_decibel(value);
+	}
+}
+
+/**
+ * Infineon MA12070P, Portable Audio DAC with Integrated Class D Speaker Driver
+ *
+ * @tparam I2cMaster I2C interface
+ *
+ * @author Raphael Lehmann
+ * @ingroup modm_driver_ma12070p
+ */
+template<typename I2cMaster>
+class Ma12070p : public ma12070p, public modm::I2cDevice<I2cMaster, 5>
+{
+public:
+	/**
+	 * Constructor.
+	 *
+	 * \param i2cAddress The I²C address depends on the AD1 and AD0 pins:
+	 *                   AD1=0, AD0=0 -> 0x20
+	 *                   AD1=0, AD0=1 -> 0x21
+	 *                   AD1=1, AD0=0 -> 0x22
+	 *                   AD1=1, AD0=1 -> 0x23
+	 */
+	Ma12070p(uint8_t i2cAddress);
+
+	/// Initialize device, call before using any other function
+	ResumableResult<bool>
+	initialize();
+
+	/// Configure I2S input format and the volume and limiter processor (VLP)
+	ResumableResult<bool>
+	configureI2sAndVlp(I2sAndVlpConfig config);
+
+	/// Set the limiter treshold for individual channels (-144db to +24db)
+	ResumableResult<bool>
+	setLimiterTreshold(quarter_decibel_t ch0l, quarter_decibel_t ch0r, quarter_decibel_t ch1l = -144, quarter_decibel_t ch1r = -144);
```

Comment:
The "treshold" is missing an "h":
```suggestion
	/// Set the limiter threshold for individual channels (-144db to +24db)
	ResumableResult<bool>
	setLimiterThreshold(quarter_decibel_t ch0l, quarter_decibel_t ch0r, quarter_decibel_t ch1l = -144, quarter_decibel_t ch1r = -144);
```

Reply:
🤦🏽